python/ola/MACAddress.py

```diff
@@ -51,19 +51,44 @@ def __hash__(self):
   def __repr__(self):
     return self.__str__()
 
-  def __cmp__(self, other):
-    if other is None:
-      return 1
-    return cmp(self.mac_address, other.mac_address)
+  def __eq__(self, other):
+    if other.__class__ is not self.__class__:
+      return False
+    return self.mac_address == other.mac_address
 
   def __lt__(self, other):
+    if other is None:
+      return False
+    if other.__class__ is not self.__class__:
+      return NotImplemented
     return self.mac_address < other.mac_address
 
-  def __eq__(self, other):
+  # These 4 can not be replaced with functools.total_ordering because of the
```

Comment:
Would this be the same that they could "be replaced with functools.total_ordering when support for 2.7 is dropped because NotImplemented is not supported by it before 3.4"? Or even when we're on Python 3.4 will functools.total_ordering still not be sufficient?

Reply:
No, we need to implement these functions ourselves because we have a special treatment applied to None that `functools.total_ordering` doesn't support.

Reply:
Is this because otherwise sorting None and a MAC you'd get stuck or something? Can we have a bit of a comment explaining why in the code please @nils-van-zuijlen ?